### PR TITLE
finish pop up flow

### PR DESCRIPTION
**Pull request by michaone007**

### CODE REVIEW

**Comment by michaone007** [comment]
code is done needs some testing with the whole workflow, plus need to talk about the timing to show the popup (if method).

**Comment by zakiva** [comment]
@michaone007 save name, gender, age, signedUpType etc. in user attributes in firebase

File: app/src/main/java/com/example/zakiva/santa/Models/SignUp.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
+package com.example.zakiva.santa.Models;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/SignUp.java]: activity should go outside of models

File: app/src/main/java/com/example/zakiva/santa/Models/SignUp.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import org.json.JSONObject;
+
+import java.util.Arrays;
+
+
+public class SignUp extends AppCompatActivity  implements View.OnClickListener, GoogleApiClient.OnConnectionFailedListener{
+
+    private static final int RC_SIGN_IN = 10;
+    private GoogleApiClient mGoogleApiClient;
+    private GoogleSignInOptions gso;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/SignUp.java]: change var name

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 3]
```diff
@@ -153,23 +157,26 @@ protected void onResume() {
...
         extras = getIntent().getExtras();
+        activityBackground = (RelativeLayout) findViewById(R.id.activityBackground);
         scoreTextView = ((TextView) findViewById(R.id.score));
         candiesTextView = ((TextView) findViewById(R.id.candies));
         expTextView = ((TextView) findViewById(R.id.exp));
         expBar = ((ProgressBar) findViewById(R.id.expBar));
-        mainIcon = ((ImageView) findViewById(R.id.mainIconScoreSection));;
-        scoreTitle = ((TextView) findViewById(R.id.scoreTitle));;
+        mainIcon = ((ImageView) findViewById(R.id.mainIconScoreSection));
+        ;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: delete

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 4]
```diff
@@ -153,23 +157,26 @@ protected void onResume() {
...
+        scoreTitle = ((TextView) findViewById(R.id.scoreTitle));
+        ;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: same

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 5]
```diff
@@ -460,4 +461,19 @@ public void shareScoreButtonClicked(View view) {
...
+
+    public void signInWindow() {
+        Handler h = new Handler();
+        h.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                changeBackgroundOpacity(220,View.VISIBLE);
+                startActivity(new Intent(Score.this, SignUp.class));
+            }
+        }, 5000);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: put number in var

**Comment by zakiva** [comment]
add cancel button to the "pop up" and disable back button
maybe need to save intent's fields
check score != -1 && not signed up yet

File: app/src/main/java/com/example/zakiva/santa/FacebookSignIn.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                try {
+                    String name, gender, ageRange;
+                    if (json != null) {
+                        gender = json.getString("gender");
+                        name = json.getString("name");
+                        ageRange = json.getString("age_range");
+                        Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getPermissions());
+                        Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getDeclinedPermissions());
+                        Infra.addGender(gender);
+                        Infra.addName(name);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/FacebookSignIn.java]: put all in one method

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -91,7 +91,12 @@
             android:configChanges="keyboardHidden|orientation|screenSize"
             android:theme="@android:style/Theme.Translucent.NoTitleBar.Fullscreen" />
         <activity android:name=".HallOfFame" />
-        <activity android:name=".PopUp"></activity>
+        <activity android:name=".PopUp" />
+        <activity
+            android:name=".SignUp"
+            android:theme="@style/Theme.AppCompat.Light.Dialog" />
+        <activity android:name=".FacebookSignIn" />
```
Comment (zakiva) [inline app/src/main/AndroidManifest.xml]: sign up

File: app/src/main/java/com/example/zakiva/santa/GoogleSignIn.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            Toast.makeText(getApplicationContext(), e.getMessage(), Toast.LENGTH_LONG).show();
+        }
+    }
+    public void getExtraInfoFromGoogle(GoogleSignInAccount account) {
+        Plus.PeopleApi.load(mGoogleApiClient, account.getId()).setResultCallback(new ResultCallback<People.LoadPeopleResult>() {
+            @Override
+            public void onResult(@NonNull People.LoadPeopleResult loadPeopleResult) {
+                Person person = loadPeopleResult.getPersonBuffer().get(0);
+                String gender = String.valueOf(person.getGender());
+                String ageRange = String.valueOf(person.getAgeRange());
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/GoogleSignIn.java]: m/f

File: app/src/main/java/com/example/zakiva/santa/Helpers/Infra.java [hunk 9]
```diff
@@ -372,6 +372,15 @@ public void onCancelled(DatabaseError databaseError) {
     public static void addSignedUpType (String type) {
         myDatabase.child("users").child(userEmail).child("attributes").child("signedUpType").setValue(type);
     }
+    public static void addAgeRange (String type) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/Infra.java]: one method here with 4 right args

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 10]
```diff
@@ -57,26 +58,32 @@
...
+
     //final VunglePub vunglePub = VunglePub.getInstance();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_score);
         initFields();
+        changeBackgroundOpacity(0, View.GONE);
+        changePlayButtonVisibility(View.VISIBLE);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: remove

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 11]
```diff
@@ -57,26 +58,32 @@
...
 
+
     //final VunglePub vunglePub = VunglePub.getInstance();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_score);
         initFields();
+        changeBackgroundOpacity(0, View.GONE);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: remove

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 12]
```diff
@@ -57,26 +58,32 @@
...
+        changePlayButtonVisibility(View.VISIBLE);
         displayScreenAccordingToScore();
+        activityBackground.getBackground().setAlpha(0);
+        signInWindow();
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: sign up
put the if here

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 13]
```diff
@@ -141,6 +147,8 @@ public void onBackPressed() {
     @Override
     protected void onRestart() {
         super.onRestart();
+        changeBackgroundOpacity(0, View.GONE);
+        changePlayButtonVisibility(View.VISIBLE);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: remove

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 14]
```diff
@@ -460,4 +465,27 @@ public void shareScoreButtonClicked(View view) {
     public void backFromScoreClicked(View view) {
         startActivity(new Intent(Score.this, Games.class));
     }
-}
+
+    public void signInWindow() {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: sign up

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 15]
```diff
@@ -460,4 +465,27 @@ public void shareScoreButtonClicked(View view) {
...
+        if (score != -1 && Storage.getStringPreferences("signedUpType",this.getApplicationContext()).equals("NONE")) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: remove if

File: app/src/main/java/com/example/zakiva/santa/Score.java [hunk 16]
```diff
@@ -460,4 +465,27 @@ public void shareScoreButtonClicked(View view) {
...
+            playButton.setClickable(false);
+            Handler h = new Handler();
+            h.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    changePlayButtonVisibility(View.GONE);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Score.java]: remove

File: app/src/main/java/com/example/zakiva/santa/SignUp.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    @Override
+    public void onBackPressed() {
+    }
+
+    public void backToScoreButton(View view) {
+        finish();
+    }
+    @Override
+    public void onResume(){
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/SignUp.java:35]: resume or restart ? or both ?

**Pull request merged**